src/cookie.rs

```diff
@@ -0,0 +1,220 @@
+    };
+    Some(::time::at_utc(::time::Timespec::new(seconds, 0)))
+}
+
+/// Represents the 'SameSite' attribute of a cookie.
+#[derive(PartialEq, Eq, Clone, Copy, Debug)]
+pub enum SameSite {
+    /// Strict same-site policy.
+    Strict,
+    /// Lax same-site policy.
+    Lax,
+}
+
+impl SameSite {
+    fn from_inner(value: cookie_crate::SameSite) -> Option<Self> {
+        match value {
+            cookie_crate::SameSite::Strict => Some(SameSite::Strict),
+            cookie_crate::SameSite::Lax => Some(SameSite::Lax),
+            cookie_crate::SameSite::None => None,
+        }
+    }
+
+    fn to_inner(value: Option<Self>) -> cookie_crate::SameSite {
+        match value {
+            Some(SameSite::Strict) => cookie_crate::SameSite::Strict,
+            Some(SameSite::Lax) => cookie_crate::SameSite::Lax,
+            None => cookie_crate::SameSite::None,
+        }
+    }
+}
+
+/// Error representing a parse failure of a 'Set-Cookie' header.
+pub struct CookieParseError(cookie::ParseError);
+
+impl<'a> fmt::Debug for CookieParseError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        self.0.fmt(f)
+    }
+}
+
+impl<'a> fmt::Display for CookieParseError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        self.0.fmt(f)
+    }
+}
+
+impl std::error::Error for CookieParseError {}
+
+/// A single HTTP cookie.
+pub struct Cookie<'a>(cookie::Cookie<'a>);
+
+impl<'a> fmt::Debug for Cookie<'a> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        self.0.fmt(f)
+    }
+}
+
+impl Cookie<'static> {
+    /// Construct a new cookie with the given name and value.
+    pub fn new<N, V>(name: N, value: V) -> Self
+    where
+        N: Into<Cow<'static, str>>,
+        V: Into<Cow<'static, str>>,
+    {
+        Self(cookie::Cookie::new(name, value))
+    }
+}
+
+impl<'a> Cookie<'a> {
+    fn parse(value: &'a ::header::HeaderValue) -> Result<Cookie<'a>, CookieParseError> {
+        std::str::from_utf8(value.as_bytes())
+            .map_err(cookie::ParseError::from)
+            .and_then(cookie::Cookie::parse)
+            .map_err(CookieParseError)
+            .map(Cookie)
+    }
+
+    pub(crate) fn into_inner(self) -> cookie::Cookie<'a> {
+        self.0
+    }
+
+    /// The name of the cookie.
+    pub fn name(&self) -> &str {
+        self.0.name()
+    }
+
+    /// Set the cookie name.
+    pub fn set_name<P: Into<Cow<'static, str>>>(&mut self, name: P) {
+        self.0.set_name(name)
+    }
+
+    /// The value of the cookie.
+    pub fn value(&self) -> &str {
+        self.0.value()
+    }
+
+    /// Set the cookie value.
+    pub fn set_value<P: Into<Cow<'static, str>>>(&mut self, value: P) {
+        self.0.set_value(value)
+    }
+
+    /// Returns true if the 'HttpOnly' directive is enabled.
+    pub fn http_only(&self) -> bool {
+        self.0.http_only().unwrap_or(false)
+    }
+
+    /// Set the 'HttpOnly' directive.
+    pub fn set_http_only(&mut self, value: bool) {
```

Comment:
Since a Client only ever *receives* `Set-Cookie` headers, and most of the attributes are from that header, is there a need to have most of these setters?

Reply:
I mostly just copied the setters from cookie.

They start making sense once there is a CookieStore::add(mycookie) method.
But we can leave them out for now.

Reply:
removed the setters for now